src/axom/inlet/tests/inlet_object.cpp

```diff
@@ -428,6 +426,114 @@ TEST(inlet_object, implicit_conversion_primitives)
   EXPECT_EQ(arr, expected_arr);
 }
 
+TEST(inlet_dict, basic_dicts)
+{
+  std::string testString =
+    "foo = { [\"key1\"] = 4, [\"key3\"] = 6, [\"key2\"] = 10}";
+  DataStore ds;
+  auto inlet = createBasicInlet(&ds, testString);
```

Comment:
Looking at this PR, I'm realizing that it is not clear to me when to use `auto` and when to use `auto&` within the `inlet` API.

Reply:
References are needed for objects owned/managed by Inlet (child Tables/Fields returned from add* functions).  It's not ideal but because the copy constructors for those objects are explicitly deleted a mistake will always result in a compile-time error instead of at runtime.